[rocprof-compute] improve interactive standalone roofline HTML (#8434)
ROCm/rocm-systems: projects/rocprofiler-compute/CHANGELOG.md

Full documentation for ROCm Compute Profiler is available at [https://rocm.docs.amd.com/projects/rocprofiler-compute/en/latest/](https://rocm.docs.amd.com/projects/rocprofiler-compute/en/latest/).

> ## ROCm Compute Profiler 3.9.0 for ROCm 7.16.0

Reviewer: Please manually check the header of this entry just before committing; we cannot rely on automerge to determine the header if we have pushed recent changes for upcoming release.

Author: Confirmed and reverted in rebase, changelog is

`## ROCm Compute Profiler 3.9.0 for ROCm 10.1.0`

matching changelog in rocprofiler-compute develop

### Added

* Added interactivity to the standalone roofline HTML written by `analyze`:
  * An *AI axis* selector chooses which memory level supplies each kernel's arithmetic intensity, or plots every level at once.
  * *Kernels* and *Bandwidth rooflines* side panels list what is drawn; click a row, a kernel point, or a bandwidth slope to isolate it, and Ctrl-click (Cmd-click on macOS) to isolate several.
  * A *Runtime shown* slider narrows the plot to the heaviest kernels by cumulative percent of GPU resident time.
  * Hover text reports a kernel's arithmetic intensity, achieved and peak throughput, percent of roofline achieved, performance limiter, dispatch count, and runtime; a roofline's hover reports its bandwidth and the compute peaks it caps against.
  * *Export PNG* saves the view currently on screen, together with a legend of the kernels and rooflines it draws.

>   * The page and chart open in the reader's system light or dark colors, and a *Dark mode* toggle overrides that for the rest of the session.

Reviewer 2: It is not the responsibility of the CHANGELOG to explain the feature - please remove this bloat.

The CHANGELOG should read like a user facing statement - its suffice to say that we have re-designed the roofline HTML to improve user experience

The analyze documentation has the responsibility of explaining the interactivity

Author: Fair, I edited the changelog to just be _Redesigned the standalone roofline HTML to improve user experience and interactivity._

> ### Changed

Reviewer 2: Seems like you need to rebase your PR on latest rocprofiler-compute to reduce unwanted diff

> * The standalone roofline HTML now drives the plot from its own toolbar and side panels rather than Plotly's built-in controls, so neither the legend nor the hover mode bar is rendered any more.

Reviewer: ```suggestion
* The standalone roofline HTML now drives the plot from its own toolbar and side panels rather than Plotly's built-in controls, so neither the legend nor the hover mode bar is rendered anymore.
```
